#!/usr/bin/env python
PACKAGE = "train"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("front_right_threshold_high", double_t, 0, "", 0.6, 0, 1);
gen.add("front_right_threshold_low", double_t, 0, "", 0.4, 0, 1);

gen.add("front_right_low", int_t, 0, "", 100, 0, 1500);
gen.add("front_right_high", int_t, 0, "", 500, 0, 1500);

gen.add("front_left_threshold_high", double_t, 0, "", 0.6, 0, 1);
gen.add("front_left_threshold_low", double_t, 0, "", 0.4, 0, 1);

gen.add("front_left_low", int_t, 0, "", 100, 0, 1500);
gen.add("front_left_high", int_t, 0, "", 500, 0, 1500);

gen.add("rear_right_threshold_high", double_t, 0, "", 0.6, 0, 1);
gen.add("rear_right_threshold_low", double_t, 0, "", 0.4, 0, 1);

gen.add("rear_right_low", int_t, 0, "", 100, 0, 1500);
gen.add("rear_right_high", int_t, 0, "", 500, 0, 1500);

gen.add("rear_left_threshold_high", double_t, 0, "", 0.6, 0, 1);
gen.add("rear_left_threshold_low", double_t, 0, "", 0.4, 0, 1);

gen.add("rear_left_low", int_t, 0, "", 100, 0, 1500);
gen.add("rear_left_high", int_t, 0, "", 500, 0, 1500);

gen.add("loop_rate", int_t, 0, "", 1000, 0, 5000);
gen.add("weight", double_t, 0, "", 0.01, 0, 1);

gen.add("vel_scale", double_t, 0, "", 1, 0, 10);

exit(gen.generate(PACKAGE, "train", "encoder_filter"));
